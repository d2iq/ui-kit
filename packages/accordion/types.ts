export type HeadingLevel = 2 | 3 | 4 | 5 | 6;
export type AccordionTitleAppearances = "default" | "danger";
