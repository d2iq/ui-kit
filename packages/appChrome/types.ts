import { BreakpointConfig } from "../shared/styles/breakpoints";
import { SpaceSizes } from "../shared/styles/styleUtils/modifiers/modifierUtils";
import { LinkProps } from "../link/types";
import { IconSize } from "../shared/types/iconSize";

type BgColor = React.CSSProperties["backgroundColor"];
type PaddingHoriz = SpaceSizes;
type PaddingVert = SpaceSizes;

export interface AppChromeTheme {
  headerBackgroundColor?: BgColor;
  headerPaddingHor?: PaddingHoriz;
  headerPaddingVert?: PaddingVert;
  iconWidth?: IconSize;
  itemActiveBackgroundColor?: BgColor;
  itemHoverBackgroundColor?: BgColor;
  sidebarBackgroundColor?: BgColor;
  sidebarHeaderPaddingHor?: PaddingHoriz;
  sidebarHeaderPaddingVert?: PaddingVert;
  sidebarWidth?: BreakpointConfig<React.CSSProperties["width"]>;
  sidebarItemPaddingHor?: PaddingHoriz;
  sidebarItemPaddingVert?: PaddingVert;
}

export interface SidebarNavItemProps extends LinkProps {
  isActive?: boolean;
  onClick?: (event?: React.SyntheticEvent<HTMLElement>) => void;
  disabled?: boolean;
}
