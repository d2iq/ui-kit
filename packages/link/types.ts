import React from "react";

export interface LinkProps {
  /**
   * A url the user will be navigated to when clicking the link. This also changes the tag to `<a>`.
   */
  url?: string;
  /**
   * If the `url` prop is set, this will open that link in a new tab.
   */
  openInNewTab?: boolean;
  children?: React.ReactNode | React.ReactNode[];
}

export type ExpandedLinkProps = LinkProps &
  Omit<React.HTMLProps<HTMLAnchorElement>, "ref">;

export type LinkComponent = React.ComponentType<
  LinkProps & React.HTMLProps<HTMLAnchorElement>
>;
