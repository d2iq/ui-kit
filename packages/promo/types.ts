export interface PromoProps {
  /** The primary banner content */
  bodyContent: React.ReactNode;
  /** A function that gets called when the banner is dismissed */
  dismissHandler?: () => void;
  /** A consise summary of the banner's message */
  headingText: string;
  /** Whether the content is on a dark background. Changes text color to be readable. */
  isDarkBackground?: boolean;
  /** A link to the graphic's image file. The graphic (typically an illustration) should be relevant to the banner's message. Ideally the graphic should not just be decorative, but it reinforces the banner's overall message. */
  graphicSrc?: string;
  /** Whether a user has opted not to see the banner */
  optOutBanner?: boolean;
  /** A function that gets called when the user checks the "Do not show again" checkbox */
  optOutHandler?: (e: React.SyntheticEvent<HTMLInputElement>) => void;
  /** The most relevant action a user can take in response to the banner's message */
  primaryAction?: React.ReactNode;
  /** An additional action a user can take in response to the banner's message. This can also just be a link to more information relevant to the banner's message. */
  secondaryAction?: React.ReactNode;
  /** The Cypress selector */
  ["data-cy"]?: string;
}

export type GradientStyle = "lightBlue" | "purple" | "oceanBlue";
export type GradientColors = [string, string, string?];
export type PromoBackgroundColor = "themeBgSecondary" | "purpleLighten5";
