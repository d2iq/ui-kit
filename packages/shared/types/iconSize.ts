export type IconSize = "xxs" | "xs" | "s" | "m" | "l" | "xl" | "xxl";
