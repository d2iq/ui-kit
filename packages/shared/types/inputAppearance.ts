export enum InputAppearance {
  Standard = "standard",
  Error = "error",
  Success = "success"
}
