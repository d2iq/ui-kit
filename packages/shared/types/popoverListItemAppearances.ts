export enum PopoverListItemAppearances {
  DANGER = "danger"
}
