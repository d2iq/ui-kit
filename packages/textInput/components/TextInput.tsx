import { cx } from "@emotion/css";
import * as React from "react";
import nextId from "react-id-generator";

import FormFieldWrapper from "../../shared/components/FormFieldWrapper";
import {
  getInputAppearanceStyle,
  inputContainer
} from "../../shared/styles/formStyles";
import { inputReset, padding } from "../../shared/styles/styleUtils";
import { flex, flexItem } from "../../shared/styles/styleUtils/layout/flexbox";
import { InputAppearance } from "../../shared/types/inputAppearance";
import { renderLabel } from "../../utilities/label";

export interface TextInputProps extends React.HTMLProps<HTMLInputElement> {
  /**
   * The HTML input type for this component.
   */
  type: "text" | "number" | "search" | "email" | "password" | "tel" | "url";
  /**
   * Sets the current appearance of the input component. This defaults to InputAppearance.Standard, but supports `InputAppearance.Error` & `InputAppearance.Success` appearances as well.
   */
  appearance: InputAppearance;
  /**
   * Sets the contents of the input label. This can be a `string` or any `ReactNode`.
   */
  inputLabel?: React.ReactNode;
  /**
   * Defaults to `true`, but can be set to `false` to visibly hide the `TextInput`'s label. The `inputLabel` should still be set even when hidden for accessibility support.
   */
  showInputLabel: boolean;
  /**
   * hintContent is text or a ReactNode that is displayed directly under the input with additional information about the expected input.
   */
  hintContent?: React.ReactNode;
  /**
   * Sets the contents for validation errors. This will be displayed below the input element. Errors are only visible when the `TextInput` appearance is also set to `InputAppearance.Error`.
   */
  errors?: React.ReactNode[];
  /**
   * Sets the text content for the tooltip that can be displayed above the input.
   */
  tooltipContent?: React.ReactNode;
}

export class TextInput<P extends TextInputProps> extends React.Component<P> {
  public static defaultProps = {
    type: "text",
    appearance: InputAppearance.Standard,
    showInputLabel: true
  };
  placeholderId = nextId("textInput-");

  public render() {
    const containerProps: { className?: string } = {};
    const appearance = this.getInputAppearance();
    const dataCy = `textInput textInput.${appearance}`;

    if (this.props.className) {
      containerProps.className = this.props.className;
    }
    return (
      <div {...containerProps} data-cy={dataCy}>
        {renderLabel({
          appearance,
          hidden: !this.props.showInputLabel,
          id: this.getId(),
          label: this.props.inputLabel,
          required: this.props.required,
          tooltipContent: this.props.tooltipContent
        })}
        {this.getInputContent()}
      </div>
    );
  }

  protected getInputAppearance(): string {
    return this.props.disabled ? "disabled" : this.props.appearance;
  }

  protected getInputContent(): React.ReactNode {
    const appearance = this.getInputAppearance();

    return (
      <FormFieldWrapper
        id={this.getId()}
        errors={this.props.errors}
        hintContent={this.props.hintContent}
      >
        {({ getValidationErrors, getHintContent, isValid, describedByIds }) => (
          <div>
            <div className={flex()}>
              {this.getInputElement(
                [
                  padding("horiz", "m"),
                  flexItem("grow"),
                  inputContainer,
                  getInputAppearanceStyle(this.getInputAppearance())
                ],
                isValid,
                describedByIds
              )}
            </div>
            <div data-cy="textInput-hintContent">
              {getHintContent}
              {appearance === InputAppearance.Error && getValidationErrors}
            </div>
          </div>
        )}
      </FormFieldWrapper>
    );
  }
  protected getInputElementProps(): TextInputProps {
    // omit props for container and that we override, otherwise pass through
    // TextInput props to input element
    const {
      className,
      hintContent,
      inputLabel,
      errors,
      id = this.placeholderId,
      ...inputElementProps
    } = this.props;

    return { ...inputElementProps, id };
  }

  protected getInputElement(
    additionalClasses: string[] = [],
    isValid: boolean,
    describedBy: string
  ) {
    const {
      value,
      showInputLabel,
      appearance,
      ...inputElementProps
    } = this.getInputElementProps();
    const textInputAppearance = this.getInputAppearance();
    const dataCy = [
      "textInput-input",
      ...(textInputAppearance &&
      textInputAppearance !== InputAppearance.Standard
        ? [`textInput-input.${textInputAppearance}`]
        : [])
    ].join(" ");
    let { onChange } = inputElementProps;
    if (onChange == null && value != null) {
      onChange = Function.prototype as (
        event: React.FormEvent<HTMLInputElement>
      ) => void;
    }

    return (
      <input
        className={cx(inputReset, ...additionalClasses)}
        type={this.props.type}
        aria-invalid={!isValid}
        aria-describedby={describedBy}
        data-cy={dataCy}
        {...{ ...inputElementProps, onChange, value }}
      />
    );
  }

  private getId(): string {
    if (typeof this.props.id === "string") {
      return this.props.id;
    }

    return this.placeholderId;
  }
}

export default TextInput;
