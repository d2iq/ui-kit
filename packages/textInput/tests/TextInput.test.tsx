import * as React from "react";
import serializer from "@emotion/jest";
import { shallow, mount, render } from "enzyme";
import toJson from "enzyme-to-json";
// tslint:disable:no-duplicate-imports
import { css, cx } from "@emotion/css";

import TextInput from "../components/TextInput";
import { InputAppearance } from "../../shared/types/inputAppearance";

expect.addSnapshotSerializer(serializer);

describe("TextInput", () => {
  it("should render all appearances with props", () => {
    Object.keys(InputAppearance).forEach(appearance => {
      const component = shallow(
        <TextInput
          id={`test.input.${InputAppearance[appearance]}`}
          inputLabel={InputAppearance[appearance]}
          appearance={InputAppearance[appearance]}
        />
      );
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  it("should render all appearances focus", () => {
    Object.keys(InputAppearance).forEach(appearance => {
      const component = mount(
        <TextInput id="test.input" appearance={InputAppearance[appearance]} />
      );
      component.find("input").simulate("focus");
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  it("should render string inputLabel", () => {
    const component = shallow(
      <TextInput id="test.input" inputLabel="Test Input" />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render node as inputLabel", () => {
    const component = shallow(
      <TextInput id="test.input" inputLabel={<span>My Test Node</span>} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should id attribute to input element", () => {
    const inputId = "test.input.0";
    const component = mount(<TextInput id={inputId} />);
    expect(component.find("input").prop("id")).toEqual(inputId);
  });

  it("should set label `for` attribute if input id set", () => {
    const inputId = "test.input.0";
    const component = shallow(
      <TextInput id={inputId} inputLabel="Test Input" />
    );
    expect(component.find("label").prop("htmlFor")).toEqual(inputId);
  });

  it("should set tabIndex on input element", () => {
    const component = mount(<TextInput id="test.input" tabIndex={2} />);
    expect(component.find("input").prop("tabIndex")).toEqual(2);
  });

  it("should call onFocus when input gains focus", () => {
    const focusFn = jest.fn();
    const component = mount(<TextInput id="test.input" onFocus={focusFn} />);
    expect(focusFn).not.toHaveBeenCalled();
    component.find("input").simulate("focus");
    expect(focusFn).toHaveBeenCalled();
  });

  it("should call onBlur when input loses focus", () => {
    const blurFn = jest.fn();
    const component = mount(<TextInput id="test.input" onBlur={blurFn} />);
    expect(blurFn).not.toHaveBeenCalled();
    component.find("input").simulate("focus");
    expect(blurFn).not.toHaveBeenCalled();
    component.find("input").simulate("blur");
    expect(blurFn).toHaveBeenCalled();
  });

  it("should call onChange when input changes", () => {
    const changeFn = jest.fn();
    const component = mount(<TextInput id="test.input" onChange={changeFn} />);
    expect(changeFn).not.toHaveBeenCalled();
    component.find("input").simulate("change");
    expect(changeFn).toHaveBeenCalled();
  });

  it("should pass className to container div", () => {
    const widthTest = css`
      width: 200px;
    `;
    const component = shallow(<TextInput id="1" className={cx(widthTest)} />);
    expect(
      component
        .find("div")
        .first()
        .props().className
    ).toEqual(widthTest);
  });

  it("should hide label if `showInputLabel` set to false", () => {
    const component = shallow(
      <TextInput
        id="input.with.hidden.label"
        inputLabel="I'm not displayed"
        showInputLabel={false}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should display validation message if set & appearance == Error", () => {
    const component = shallow(
      <TextInput
        id="input.error.with.message"
        inputLabel="Error Message Test"
        appearance={InputAppearance.Error}
        errors={["This is an error message", "this is a second error message"]}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should not display validation message if set & appearance == Success", () => {
    const component = shallow(
      <TextInput
        id="input.success.without.message"
        inputLabel="No Error Message Test"
        appearance={InputAppearance.Success}
        errors={["This is an error message"]}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should display icon with tooltip if tooltipText is set", () => {
    const component = shallow(
      <TextInput
        id="input.with.tooltip"
        inputLabel="Tooltip Message Test"
        tooltipContent="Example Tooltip"
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should generate an ID if one is not passed", () => {
    const component = render(<TextInput inputLabel="Test Input" />);
    // const textInputId = component.find("input").prop("id");
    // console.log(`textInputId: ${textInputId}`);
    expect(component.find("input").prop("id")).toBeDefined();
  });
});
