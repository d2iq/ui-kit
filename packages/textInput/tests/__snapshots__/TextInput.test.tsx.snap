// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should display icon with tooltip if tooltipText is set 1`] = `
.emotion-0 {
  -webkit-align-items: flex;
  -webkit-box-align: flex;
  -ms-flex-align: flex;
  align-items: flex;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
}

.emotion-0>div {
  width: auto;
}

.emotion-1 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

.emotion-2 {
  margin-left: 8px;
  margin-bottom: 4px;
}

<div
  data-cy="textInput textInput.standard"
>
  <div
    className="emotion-0"
  >
    <label
      className="emotion-1"
      data-cy="textInput-label"
      htmlFor="input.with.tooltip"
    >
      Tooltip Message Test
    </label>
    <span
      className="emotion-2"
    >
      <Tooltip
        id="labelTooltip-input.with.tooltip"
        maxWidth={200}
        trigger={
          <Icon
            color="#AEB0B4"
            shape="system-circle-question"
            size="xs"
          />
        }
      >
        Example Tooltip
      </Tooltip>
    </span>
  </div>
  <FormFieldWrapper
    id="input.with.tooltip"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;

exports[`TextInput should display validation message if set & appearance == Error 1`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
  color: var(--themeError, #EB293A);
  fill: var(--themeError, #EB293A);
}

<div
  data-cy="textInput textInput.error"
>
  <label
    className="emotion-0"
    data-cy="textInput-label"
    htmlFor="input.error.with.message"
  >
    Error Message Test
  </label>
  <FormFieldWrapper
    errors={
      Array [
        "This is an error message",
        "this is a second error message",
      ]
    }
    id="input.error.with.message"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;

exports[`TextInput should hide label if \`showInputLabel\` set to false 1`] = `
.emotion-0 {
  border: 0;
  clip: rect(0 0 0 0);
  position: absolute;
  overflow: hidden;
  margin: -1px;
  padding: 0;
  width: 1px;
  height: 1px;
}

<div
  data-cy="textInput textInput.standard"
>
  <label
    className="emotion-0"
    data-cy="textInput-label"
    htmlFor="input.with.hidden.label"
  >
    I'm not displayed
  </label>
  <FormFieldWrapper
    id="input.with.hidden.label"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;

exports[`TextInput should not display validation message if set & appearance == Success 1`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

<div
  data-cy="textInput textInput.success"
>
  <label
    className="emotion-0"
    data-cy="textInput-label"
    htmlFor="input.success.without.message"
  >
    No Error Message Test
  </label>
  <FormFieldWrapper
    errors={
      Array [
        "This is an error message",
      ]
    }
    id="input.success.without.message"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;

exports[`TextInput should render all appearances focus 1`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

.emotion-1 {
  -webkit-align-items: flex;
  -webkit-box-align: flex;
  -ms-flex-align: flex;
  align-items: flex;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  background: none;
  border: 0;
  box-shadow: none;
  padding: 0;
  padding-left: 16px;
  padding-right: 16px;
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  height: 36px;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-color: var(--themeBorder, #DADDE2);
}

.emotion-2:focus {
  outline: 0;
}

.emotion-2::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input {
  font-size: inherit;
  color: inherit;
}

.emotion-2 input::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:focus {
  background-color: rgba(125,88,255,0.05);
  border-color: var(--themeBrandPrimary, #7D58FF);
}

<TextInput
  appearance="standard"
  id="test.input"
  showInputLabel={true}
  type="text"
>
  <div
    data-cy="textInput textInput.standard"
  >
    <label
      className="emotion-0"
      data-cy="textInput-label"
      htmlFor="test.input"
    />
    <FormFieldWrapper
      id="test.input"
    >
      <div>
        <div
          className="emotion-1"
        >
          <input
            aria-describedby=""
            aria-invalid={false}
            className="emotion-2"
            data-cy="textInput-input"
            id="test.input"
            type="text"
          />
        </div>
        <div
          data-cy="textInput-hintContent"
        />
      </div>
    </FormFieldWrapper>
  </div>
</TextInput>
`;

exports[`TextInput should render all appearances focus 2`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
  color: var(--themeError, #EB293A);
  fill: var(--themeError, #EB293A);
}

.emotion-1 {
  -webkit-align-items: flex;
  -webkit-box-align: flex;
  -ms-flex-align: flex;
  align-items: flex;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  background: none;
  border: 0;
  box-shadow: none;
  padding: 0;
  padding-left: 16px;
  padding-right: 16px;
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  height: 36px;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-color: var(--themeError, #EB293A);
}

.emotion-2:focus {
  outline: 0;
}

.emotion-2::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input {
  font-size: inherit;
  color: inherit;
}

.emotion-2 input::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:focus {
  background-color: rgba(235,41,58,0.05);
}

<TextInput
  appearance="error"
  id="test.input"
  showInputLabel={true}
  type="text"
>
  <div
    data-cy="textInput textInput.error"
  >
    <label
      className="emotion-0"
      data-cy="textInput-label"
      htmlFor="test.input"
    />
    <FormFieldWrapper
      id="test.input"
    >
      <div>
        <div
          className="emotion-1"
        >
          <input
            aria-describedby=""
            aria-invalid={false}
            className="emotion-2"
            data-cy="textInput-input textInput-input.error"
            id="test.input"
            type="text"
          />
        </div>
        <div
          data-cy="textInput-hintContent"
        />
      </div>
    </FormFieldWrapper>
  </div>
</TextInput>
`;

exports[`TextInput should render all appearances focus 3`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

.emotion-1 {
  -webkit-align-items: flex;
  -webkit-box-align: flex;
  -ms-flex-align: flex;
  align-items: flex;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  background: none;
  border: 0;
  box-shadow: none;
  padding: 0;
  padding-left: 16px;
  padding-right: 16px;
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  height: 36px;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-color: var(--themeSuccess, #14C684);
}

.emotion-2:focus {
  outline: 0;
}

.emotion-2::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input {
  font-size: inherit;
  color: inherit;
}

.emotion-2 input::-webkit-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::-moz-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input:-ms-input-placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2 input::placeholder {
  color: var(--themeTextColorPrimary, #1B2029);
  opacity: 0.4;
}

.emotion-2:focus {
  background-color: rgba(20,198,132,0.05);
}

<TextInput
  appearance="success"
  id="test.input"
  showInputLabel={true}
  type="text"
>
  <div
    data-cy="textInput textInput.success"
  >
    <label
      className="emotion-0"
      data-cy="textInput-label"
      htmlFor="test.input"
    />
    <FormFieldWrapper
      id="test.input"
    >
      <div>
        <div
          className="emotion-1"
        >
          <input
            aria-describedby=""
            aria-invalid={false}
            className="emotion-2"
            data-cy="textInput-input textInput-input.success"
            id="test.input"
            type="text"
          />
        </div>
        <div
          data-cy="textInput-hintContent"
        />
      </div>
    </FormFieldWrapper>
  </div>
</TextInput>
`;

exports[`TextInput should render all appearances with props 1`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

<div
  data-cy="textInput textInput.standard"
>
  <label
    className="emotion-0"
    data-cy="textInput-label"
    htmlFor="test.input.standard"
  >
    standard
  </label>
  <FormFieldWrapper
    id="test.input.standard"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;

exports[`TextInput should render all appearances with props 2`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
  color: var(--themeError, #EB293A);
  fill: var(--themeError, #EB293A);
}

<div
  data-cy="textInput textInput.error"
>
  <label
    className="emotion-0"
    data-cy="textInput-label"
    htmlFor="test.input.error"
  >
    error
  </label>
  <FormFieldWrapper
    id="test.input.error"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;

exports[`TextInput should render all appearances with props 3`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

<div
  data-cy="textInput textInput.success"
>
  <label
    className="emotion-0"
    data-cy="textInput-label"
    htmlFor="test.input.success"
  >
    success
  </label>
  <FormFieldWrapper
    id="test.input.success"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;

exports[`TextInput should render node as inputLabel 1`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

<div
  data-cy="textInput textInput.standard"
>
  <label
    className="emotion-0"
    data-cy="textInput-label"
    htmlFor="test.input"
  >
    <span>
      My Test Node
    </span>
  </label>
  <FormFieldWrapper
    id="test.input"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;

exports[`TextInput should render string inputLabel 1`] = `
.emotion-0 {
  display: block;
  padding-top: 0;
  margin-top: 0;
  margin-bottom: 4px;
  font-weight: 500;
}

<div
  data-cy="textInput textInput.standard"
>
  <label
    className="emotion-0"
    data-cy="textInput-label"
    htmlFor="test.input"
  >
    Test Input
  </label>
  <FormFieldWrapper
    id="test.input"
  >
    <Component />
  </FormFieldWrapper>
</div>
`;
